routes/project.js

```diff
@@ -250,18 +251,37 @@ router.get('/:branch/*.xslx', function(req, res, next) {
 	res.send('I am xslx for ' + req.params[0]);
 });
 
-router.get('/:branch/history', function(req, res, next) {
+router.get('/:branch/history/:page?', function(req, res, next) {
 	var branchR = req.params.branch || req.params.commit;
 	var branch = branchR || 'master';
 	branch = branch.replace(/^~/, ''); //remove leading ~ when we match with branch with command
 	
 	var area = req.params.area;
 	var project = req.params.project;
+	var itemsPerPage = 15;
+	var page = parseInt(req.params.page || 1);
 
 	var title = area + '/' + project + (branchR ? '/' + branchR : '')
 	history(req, branch, function(err, commits) {
 		if (err) return next(err);
-		res.render('history', {commits: commits, title: title, errors: []})
+
+		var maxPages = Math.ceil(commits.length / itemsPerPage);
+		page = page < 1 ? 1 : page;
+		page = page > maxPages ? maxPages : page;
+
+		function slicer(items){
+			var offset = (itemsPerPage*(page-1));
+			if(items.length > offset){
+				return items.slice(offset);
+			}
+			return items;
+		}
+
+		res.render('history', {	commits: slicer(commits),
+								title: title,
```

Comment:
This is not how you should do paging for a github repository. As it is currently written, you would need the history function to scan the full history of the repository every time. A better option would be to just render the url of the next button as a link the history of the last commit hash from commits (and drop the last commit from the display - unless you like the one commit overlap).